extensions/wikia/TwitterTag/TwitterTagController.class.php

```diff
@@ -40,28 +40,30 @@ class TwitterTagController extends WikiaController {
 		// Favourites Timeline, but there's no (documented) parameter to force any of them.
 	];
 
-	static public function onParserFirstCallInit( Parser $parser ) {
-		$parser->setHook( self::PARSER_TAG_NAME, [ new static(), 'parseTag' ] );
-		return true;
+	public function getTagName() {
+		return self::PARSER_TAG_NAME;
 	}
 
-	/**
-	 * Parses the twitter tag. Checks to ensure the required attributes are there.
-	 * Then constructs the HTML after seeing which attributes are in use.
-	 *
-	 * @return string
-	 */
-	public function parseTag( $input, array $args, Parser $parser, PPFrame $frame ) {
-		if ( empty( $args['widget-id'] ) ) {
-			return '<strong class="error">' . wfMessage( 'twitter-tag-widget-id' )->parse() . '</strong>';
-		}
+	protected function getAttributesAllowed() {
+		return array_keys( self::TAG_PERMITTED_ATTRIBUTES );
+	}
+
+	protected function getErrorOutput( $errorMessages ) {
+		return $this->sendRequest(
+			'TwitterTagController',
+			'twitterTagError',
+			[ 'errorMessages' => $errorMessages ]
+		);
+	}
 
-		$attributes = $this->prepareAttributes( $args, self::TAG_PERMITTED_ATTRIBUTES );
+	protected function getSuccessOutput( $args ) {
+		$attributes = $this->buildTagAttributes( self::TAG_PERMITTED_ATTRIBUTES, $args, 'data' );
 		$attributes['href'] = self::TWITTER_BASE_URL;
+
 		// data-wikia-widget attribute is searched for by Mercury
 		$attributes['data-wikia-widget'] = self::PARSER_TAG_NAME;
 
-		if ( ( new WikiaIFrameTagBuilderHelper() )->isMobileSkin() ) {
```

Comment:
Oh, we'll have to sync it somehow with the Flite tag/widget @fraszczakszymon was working on recently: https://github.com/Wikia/app/pull/8806/files#diff-08ba9d6c112d52c6530b470657257f75R24